tests/System.Buffers.Primitives.Tests/BufferReferenceTesting.cs

```diff
@@ -6,17 +6,30 @@ namespace System.Buffers.Tests
 {
     public class BufferReferenceTests
     {
-        public static void TestOwnedBuffer(Func<OwnedBuffer<byte>> create)
+        public static void TestOwnedBuffer(Func<OwnedBuffer<byte>> create, Action<OwnedBuffer<byte>> dispose = null)
```

Comment:
When block is allocated from MemoryPool it has to be properly returned, otherwise `Assert` would fire during finalization crashing test process.